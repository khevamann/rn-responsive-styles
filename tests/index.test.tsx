import * as React from 'react'
import { ScaledSize } from 'react-native'
import * as renderer from 'react-test-renderer'

import { cleanup } from '@testing-library/react-native'
import Sample from './Sample'

afterEach(cleanup)

const mockDimensions = ({ width }: Pick<ScaledSize, 'width'>) => {
  jest.resetModules()
  jest.doMock('react-native/Libraries/Utilities/Dimensions', () => ({
    get: jest.fn().mockReturnValue({ width }),
    addEventListener: jest.fn(),
  }))
}
describe('tests that device sizes render appropriately', () => {
  test('tests that extra small devices render correctly', () => {
    mockDimensions({ width: 414 })
    const tree = renderer.create(<Sample />).toJSON()
    expect(tree).toMatchSnapshot('extrasmall')
  })
  test('tests that small devices render correctly', () => {
    mockDimensions({ width: 580 })
    const tree = renderer.create(<Sample />).toJSON()
    expect(tree).toMatchSnapshot('small')
  })
  test('tests that medium devices render correctly', () => {
    mockDimensions({ width: 770 })
    const tree = renderer.create(<Sample />).toJSON()
    expect(tree).toMatchSnapshot('medium')
  })
  test('tests that large devices render correctly', () => {
    mockDimensions({ width: 1000 })
    const tree = renderer.create(<Sample />).toJSON()
    expect(tree).toMatchSnapshot('large')
  })
  test('tests that extra large devices render correctly', () => {
    mockDimensions({ width: 1300 })
    const tree = renderer.create(<Sample />).toJSON()
    expect(tree).toMatchSnapshot('extralarge')
  })
})
