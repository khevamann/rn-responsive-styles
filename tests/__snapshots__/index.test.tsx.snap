// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests that device sizes render appropriately tests that extra large devices render correctly: extralarge 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "purple",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "orange",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
        },
        Array [
          Object {
            "color": "black",
          },
          Object {
            "fontSize": 30,
          },
        ],
      ]
    }
  >
    Device Size: 
    extralarge
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "white",
        },
        Array [
          Object {
            "color": "black",
          },
          Object {
            "fontSize": 30,
          },
        ],
      ]
    }
  >
    Color: 
    orange
  </Text>
</View>
`;

exports[`tests that device sizes render appropriately tests that extra small devices render correctly: extrasmall 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "purple",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
        },
        Object {
          "fontSize": 20,
        },
      ]
    }
  >
    Device Size: 
    extrasmall
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "white",
        },
        Object {
          "fontSize": 20,
        },
      ]
    }
  >
    Color: 
    purple
  </Text>
</View>
`;

exports[`tests that device sizes render appropriately tests that large devices render correctly: large 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "purple",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "blue",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
        },
        Object {
          "fontSize": 30,
        },
      ]
    }
  >
    Device Size: 
    large
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "white",
        },
        Object {
          "fontSize": 30,
        },
      ]
    }
  >
    Color: 
    blue
  </Text>
</View>
`;

exports[`tests that device sizes render appropriately tests that medium devices render correctly: medium 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "purple",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "green",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
        },
        Object {
          "fontSize": 20,
        },
      ]
    }
  >
    Device Size: 
    medium
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "white",
        },
        Object {
          "fontSize": 20,
        },
      ]
    }
  >
    Color: 
    green
  </Text>
</View>
`;

exports[`tests that device sizes render appropriately tests that small devices render correctly: small 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "purple",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
        },
        Object {
          "fontSize": 20,
        },
      ]
    }
  >
    Device Size: 
    small
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "white",
        },
        Object {
          "fontSize": 20,
        },
      ]
    }
  >
    Color: 
    red
  </Text>
</View>
`;
